src/flint/fmpz_mat.jl

```diff
@@ -1554,7 +1554,11 @@ function _solve_dixon(a::ZZMatrix, b::ZZMatrix)
 end
 
 #XU = B. only the upper triangular part of U is used
-function _solve_triu_left(U::ZZMatrix, b::ZZMatrix)
+function AbstractAlgebra._solve_triu_left(U::ZZMatrix, b::ZZMatrix; side::Symbol = :left)
+  if side == :right
+    return AbstractAlgebra._solve_triu(U, b; side)
+  end
```

Comment:
uh... why does `_solve_triu_left` have a `side` parameter to begin with?!?

As far as I can tell *nothing* calls `_solve_triu_left` except one test case in Nemo.jl. So we could also just remove it?

Reply:
No. solve_triu left is essential for the stuff in Hecke. The solve right is not used (currently)

Reply:
I agree with @fingolfin that `_solve_triu` should have the `side` kwarg and dispatch to different methods depending on it. `_solve_triu_left` should not have to deal with these things.